tensorflow/core/kernels/resize_bilinear_op_gpu.cu.cc

```diff
@@ -278,23 +344,55 @@ struct ResizeBilinear<GPUDevice, T> {
     const int total_count = batch * out_height * out_width * channels;
     if (total_count == 0) return;
 
-    GpuLaunchConfig config = GetCudaLaunchConfig(total_count, d);
-    if (half_pixel_centers) {
-      TF_CHECK_OK(CudaLaunchKernel(
-          ResizeBilinearKernel<T>, config.block_count, config.thread_per_block,
-          0, d.stream(), config.virtual_thread_count, images.data(),
-          height_scale, width_scale, batch, in_height, in_width, channels,
-          out_height, out_width, output.data()));
-    } else {
+    // ResizeBilinearKernel_faster is 30 ~ 50% faster than ResizeBilinearKernel
+    // but can only be used when channels is multiple of 4 and size of input
+    // elemnt is the same as float
+    if (channels % 4 == 0 && sizeof(float) == sizeof(T) && half_pixel_centers) {
+      // since each thread reads 16 bytes, and we need at most 8 of such threads
+      // to make the full use of 128 bytes of global memroy read & write
+      const int channel_per_thread = 16 / sizeof(float);
+
+      // since each global memroy read from L1 cahce is 128 bytes, and each thread
+      // reads 16 bytes, we need 8 threads to fully coalesce 128 bytes of read & store
+      const int max_num_channel_thread = 8;
+
+      // number of threads that will iterate through the channel dimension
+      const int num_channel_thread = std::min(max_num_channel_per_thread,
+                                              num_channels/channel_per_thread);
+
+      GpuLaunchConfig config = GetCudaLaunchConfig(out_height * out_width *
+                                                   num_channel_thread, d);
+
       TF_CHECK_OK(CudaLaunchKernel(
-          LegacyResizeBilinearKernel<T>, config.block_count,
-          config.thread_per_block, 0, d.stream(), config.virtual_thread_count,
-          images.data(), height_scale, width_scale, batch, in_height, in_width,
-          channels, out_height, out_width, output.data()));
+        ResizeBilinearKernel_faster<T, channel_per_thread>,
```

Comment:
Just an idea: Have you considered assigning a function pointer and config variable in these if/else blocks and then have a common CudaLaunchKernel call? If the arguments are otherwise the same, that makes it a little less verbose.